src/lib/web3.ts

```diff
@@ -0,0 +1,159 @@
+import { Wallet, ethers } from 'ethers';
+import axios from 'axios';
+
+import { getWallet } from './wallet';
+import { getTenderlySettings } from './keys';
+import { exitWithMessage, logs, networks, warning } from './constants';
+import {
+  spinnerError,
+  spinnerSuccess,
+  stopSpinner,
+  updateSpinnerText,
+} from './spinners';
+import {
+  ContractArtifact,
+  ContractDeploymentResult,
+  ForkResult,
+  Network,
+  TenderlySettings,
+} from 'src/types';
+
+/**
+ * Deploys a contract to a specified network and returns the contract address and transaction hash.
+ *
+ * @param {string} rpc - The RPC URL of the network to deploy the contract to.
+ * @param {ContractArtifact} contractBuild - The build of the contract to be deployed.
+ * @returns {Promise<ContractDeploymentResult>} A promise that resolves to an object containing the contract address and transaction hash.
+ * @throws Will throw an error if the contract deployment fails.
+ */
+export const deployContract = async (
+  rpc: string,
+  contractBuild: ContractArtifact,
+): Promise<ContractDeploymentResult> => {
+  const wallet = await getWallet();
+  const provider = new ethers.providers.JsonRpcProvider(rpc);
+  let address, txHash;
+  const pluginSetup = new ethers.ContractFactory(
+    contractBuild.abi,
+    contractBuild.bytecode,
+    wallet?.connect(provider),
+  );
+
+  try {
+    updateSpinnerText('Deploying contract...');
+    const instance = await pluginSetup.deploy();
+    await instance.deployed();
+    address = instance.address;
+    txHash = instance.deployTransaction.hash;
+    stopSpinner();
+  } catch (error) {
+    spinnerError('Failed to deploy contract');
+    console.error(error);
+    process.exit(1);
+  }
+
+  return { address, txHash };
+};
+
+/**
+ * Forks a network using the Tenderly API and provides the fork details.
+ *
+ * @param {string} networkId - The ID of the network to be forked. Defaults to '1'.
+ * @returns {Promise<ForkResult>} A promise that resolves to an object containing the forkId, rpcUrl, forkUrl, and forkProvider.
+ * @throws Will throw an error if the Tenderly settings are not found.
+ */
+export const tenderlyFork = async (networkId = '1'): Promise<ForkResult> => {
+  const { tenderlyKey, tenderlyUsername, tenderlyProject } =
+    (await getTenderlySettings()) as TenderlySettings;
+
+  // If any of the tenderly settings are missing, exit
+  if (!tenderlyKey || !tenderlyUsername || !tenderlyProject)
+    exitWithMessage('Tenderly settings not found');
+
+  const fork = await axios.post(
+    `https://api.tenderly.co/api/v1/account/${tenderlyUsername}/project/${tenderlyProject}/fork`,
+    { network_id: networkId },
+    { headers: { 'X-Access-Key': tenderlyKey as string } },
```

Comment:
Did linter suggest the `} }`? Usually I see it as `}}`

Reply:
yep, its the oppisite of this : {